src/httprpc.cpp

```diff
@@ -184,6 +191,23 @@ static bool HTTPReq_JSONRPC(HTTPRequest* req, const std::string &)
         // Set the URI
         jreq.URI = req->GetURI();
 
+#ifdef ENABLE_WALLET
+        if (walletName.empty()) {
```

Comment:
> That would be less clean when calling in from GUI, tests, etc.

I can see how deriving the wallet pointer in GetWalletForJSONRPCRequest could make gui & test code messier depending on how it was written, but if you provided an inverse function that set up the request object given the wallet name or pointer, I think that would keep things clean and isolated.

Anyway, this is just a thought. It looks like if you wanted you could get rid of a lot of ifdefs even keeping the pointer member.

> But maybe a void* would make sense on the class?

As long as there is a pointer member, I don't think making it void has advantages over using the forward declaration. Changing it to void just throws away type information and requires you to add casts, without letting you do anything that wasn't possible otherwise.